packages/styled-components/src/models/ComponentStyle.js

```diff
@@ -41,19 +44,27 @@ export default class ComponentStyle {
     const { componentId } = this;
 
     if (this.isStatic) {
-      if (!styleSheet.hasNameForId(componentId, componentId)) {
-        const cssStatic = flatten(this.rules, executionContext, styleSheet).join('');
+      if (this.staticRulesId && styleSheet.hasNameForId(componentId, this.staticRulesId)) {
+        return this.staticRulesId;
+      }
+
+      const cssStatic = flatten(this.rules, executionContext, styleSheet).join('');
+      const name = generateName(phash(this.baseHash, cssStatic) >>> 0);
```

Comment:
I like how this still preserves the phash optimisation 👍 but given that in prod we know that this can never change and this path of isStatic will never be taken, isn't it safe just to generate a name from the base hash itself?

I know this isn't going to provide much entropy so we could hash the length of cssStatic itself as well. I'm just concerned that this might regress on some of the performance gains we've made

Reply:
like `phash(this.baseHash, 1)`? or just basehash by itself?

Reply:
Like ‘phash(this.baseHash, cssStatic.length)’